RFC-0003-arrow-flight-connector.md

```diff
@@ -0,0 +1,83 @@
+# **RFC0003 for Presto**
+
+## Arrow Flight connector template in Presto
+
+Proposers
+
+* Elbin Pallimalil
+
+## [Related Issues]
+
+https://github.com/prestodb/presto/issues/22729
+
+## Summary
+
+Create a base Arrow Flight module in Presto which can be extended to create an Arrow Flight connector.
+
+## Background
+
+Apache Arrow Flight is a new general-purpose client-server framework that simplifies high performance transport of large datasets over network interfaces. https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/
+
+Presto currently supports different datasources through connectors for each type of datasource. The connectors can use JDBC or some other protocol for datasource connection. If we use Flight client to connect to the different datasources we can improve the performance of the datasource connection. Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.
+
+## Proposed Implementation
+
+A template for Flight connector will be created as `presto-base-arrow-flight` module. Since each implementation of flight server will be different, this base module can be extended to create a flight connector that supports a particular implementation. The template will support only read operations.
+
+### High level design
+
+#### Proposed design
+
+![Design diagram](arrow-flight-connector/Presto-flight-client.drawio.png)
+
+The Arrow Flight libraries provide a development framework for implementing a service that can discover metadata and send and receive data streams. Flight APIs are used to retrieve the data from the selected table. The steps below need to be followed. 
+
+1. Create a Flight Client using the host, port, ssl information of the Flight service where the route is opened. 
```

Comment:
Does this mean that each task will be creating a Flight client? There is some cost for doing that, I believe mostly the gRPC channel, so it might be worth looking into if there is a benefit to reuse or share clients rather than creating many new ones. @lidavidm might have some insight on this.

Reply:
Each gRPC channel is 0 or more actual connections, so yes, if you can reuse clients it would save some resources (unless you're maxing out the bandwidth in which case you'd probably want separate clients)

Reply:
The current design is such that a FlightClient will be created for a new query and then will be closed after use. Are you proposing to keep this FlightClient open?

Reply:
It is not strictly necessary but it would be good to keep a pool of clients for frequently used destinations. Of course that depends on the workload. I think this is more an implementation detail though.

Reply:
Added some details on how a connection pool can be maintained of Flight Clients.